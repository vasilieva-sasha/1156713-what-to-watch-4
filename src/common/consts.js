export const TIMEOUT = 1000;

export const SIMILAR_FILMS_AMOUNT_SHOW = 4;

export const ALL_GENRES = `All genres`;

export const MAX_GENRES = 9;

export const FILMS_SHOW_AMOUNT = 8;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const CurrentPage = {
  MAIN: `MAIN`,
  INFO: `INFO`,
  LOGIN: `LOGIN`,
  REVIEW: `REVIEW`,
  MYLIST: `MYLIST`
};

export const AppRoute = {
  MAIN: `/`,
  SIGN_IN: `/login`,
  MYLIST: `/mylist`,
  FILM: `/films`,
  ADD_REVIEW: `/review`,
  PLAYER: `/player/`
};

export const RATING = 5;

export const Rating = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`
};

export const Score = {
  BAD: 3,
  NORMAL: 5,
  VERY_GOOD: 8,
  AWESOME: 10
};

export const Comment = {
  MIN: 50,
  MAX: 400
};

export const MINUTES_IN_HOUR = 60;
