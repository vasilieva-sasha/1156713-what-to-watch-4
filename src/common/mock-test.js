export const Movie = {
  TITLE: `The Grand Budapest Hotel`,
  GENRE: `Drama`,
  DATE: 2014
};

export const TITLES = [`Fantastic Beasts: The Crimes of Grindelwald`, `Bohemian Rhapsody`, `Macbeth`, `Aviator`,
  `We need to talk about Kevin`, `What We Do in the Shadows`, `Revenant`, `Johnny English`, `Shutter Island`,
  `Pulp Fiction`, `No Country for Old Men`, `Snatch`, `Moonrise Kingdom`, `Seven Years in Tibet`, `Midnight Special`,
  `War of the Worlds`, `Dardjeeling Limited`, `Orlando`, `Mindhunter`, `The Grand Budapest Hotel`];
