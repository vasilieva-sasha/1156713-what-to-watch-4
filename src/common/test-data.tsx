import {Film} from "../types";

export const noop = () => {
  // Mock function for test props
};

export const film: Film = {
  id: 3,
  title: `a`,
  genre: `Drama`,
  releaseDate: 2018,
  runtime: 200,
  poster: `fantastic-beasts-the-crimes-of-grindelwald`,
  posterInfo: `posters/placeimg_218_327_animals`,
  background: `background/placeimg_1000_424_animals`,
  backgroundColor: `rgba`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: {
    score: 8.9,
    count: 240
  },
  text: ``,
  director: `Wes Andreson`,
  actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  isFavorite: false
};

export const films: Array<Film> = [
  {
    id: 1,
    title: `a`,
    genre: `Drama`,
    releaseDate: 2018,
    runtime: 200,
    poster: `fantastic-beasts-the-crimes-of-grindelwald`,
    posterInfo: `posters/placeimg_218_327_animals`,
    background: `background/placeimg_1000_424_animals`,
    backgroundColor: `rgba`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: {
      score: 8.9,
      count: 240
    },
    text: ``,
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    isFavorite: false
  },
  {
    id: 2,
    title: `b`,
    genre: `Drama`,
    releaseDate: 2018,
    runtime: 200,
    poster: `fantastic-beasts-the-crimes-of-grindelwald`,
    posterInfo: `posters/placeimg_218_327_animals`,
    background: `background/placeimg_1000_424_animals`,
    backgroundColor: `rgba`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: {
      score: 8.9,
      count: 240
    },
    text: ``,
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    isFavorite: true
  }
];
