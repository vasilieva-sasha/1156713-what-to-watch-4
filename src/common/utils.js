import {ALL_GENRES, Rating, Score, MINUTES_IN_HOUR} from "./consts";

export const makeLetterUppercase = (string) => {
  return string[0].toUpperCase() + string.slice(1);
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getGenres = (films) => {
  return films.reduce((filmGenres, film) => {
    if (!filmGenres.includes(film.genre)) {
      filmGenres.push(film.genre);
    }
    return filmGenres;
  }, [ALL_GENRES]);
};

export const getRatingLevel = (rating) => {
  switch (true) {
    case rating <= Score.BAD:
      return Rating.BAD;
    case rating > Score.BAD && rating <= Score.NORMAL:
      return Rating.NORMAL;
    case rating > Score.NORMAL && rating <= Score.VERY_GOOD:
      return Rating.GOOD;
    case rating < Score.VERY_GOOD && rating < Score.AWESOME:
      return Rating.VERY_GOOD;
    case rating === Score.AWESOME:
      return Rating.AWESOME;
  }
  return null;
};

export const getFilmDuration = (runtime) => {
  const hours = Math.floor(runtime / MINUTES_IN_HOUR);
  const minutes = runtime % MINUTES_IN_HOUR;
  const filmDuration = `${hours}h ${minutes}m`;
  return filmDuration;
};
