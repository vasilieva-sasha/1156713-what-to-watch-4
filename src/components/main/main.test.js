import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import Main from "./main";
import {Movie} from "../../common/mock-test";
import {getGenres} from "../../common/utils";

const films = [{
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  genre: `Drama`,
  releaseDate: 2018,
  poster: `fantastic-beasts-the-crimes-of-grindelwald`,
  posterInfo: `posters/placeimg_218_327_animals`,
  background: `background/placeimg_1000_424_animals`,
  preview: ``,
  rating: {
    score: `8,9`,
    level: `Very good`,
    count: 240
  },
  text: [`text`, `text`],
  director: `Wes Andreson`,
  actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`]
}];

const mockStore = configureStore([]);

describe(`MainComponent`, () => {
  it(`Main correct render`, () => {
    const store = mockStore({
      genre: `All genres`,
      genres: getGenres(films),
      films,
      filteredFilms: films
    });

    const tree = renderer
    .create(
        <Provider store={store}>
          <Main
            title={Movie.TITLE}
            genre={Movie.GENRE}
            date={Movie.DATE}
            genres={store.genres}
            films={store.filteredFilms}
            activeGenre={store.genre}
            onCardClick={() => {}}
          />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
