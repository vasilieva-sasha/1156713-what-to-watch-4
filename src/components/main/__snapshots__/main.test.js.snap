// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent Main correct render 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        src="background/placeimg_1000_424_animals"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="#"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Fantastic Beasts: The Crimes of Grindelwald poster"
            height="327"
            src="posters/placeimg_218_327_animals"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2018
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/films/1/player"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/1"
            onClick={[Function]}
          >
            <video
              poster="fantastic-beasts-the-crimes-of-grindelwald"
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </a>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
      </div>
      
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
